x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -193,8 +194,9 @@ export class EnterpriseSearchPlugin implements Plugin {
     try {
       this.data = await http.get('/internal/enterprise_search/config_data');
       this.hasInitialized = true;
-    } catch {
+    } catch (e) {
       this.data.errorConnecting = true;
+      this.data.errorConnectingMessage = `${e.res.status} ${e.message}`;
```

Comment:
If you're trying to trace your way through this PR, this is where it all starts. The status and message from the error are captured here, and are drilled through props down through the various entry points (AppSearch, WorkplaceSearch, and EnterpriseSearch) down to the `ErrorStatePrompt` component, where they are rendered to the user.